.github/workflows/assess-file-changes.yml

```
@@ -0,0 +1,53 @@
+name: Assess file changes
+
+on:
+  workflow_call:
+    # Map the workflow outputs to job outputs
+    outputs:
+      SOURCE_CHANGED:
+        description: "Whether the files under /src/ were changed."
+        value: ${{ jobs.build.outputs.SOURCE_CHANGED }}
+      CHANGELOG_UPDATED:
+        description: "Whether or the CHANGELOG.md file was updated."
+        value: ${{ jobs.build.outputs.CHANGELOG_UPDATED }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    # Map the job outputs to step outputs
+    outputs:
+      SOURCE_CHANGED: ${{ steps.assess-changes.outputs.SOURCE_CHANGED }}
+      CHANGELOG_UPDATED: ${{ steps.assess-changes.outputs.CHANGELOG_UPDATED }}
+
+    name: Test changed-files
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v29.0.4
+
+      - name: Assess Source Code Changes
+        id: assess-changes
+        run: |
+          echo "SOURCE_CHANGED=false" >> $GITHUB_OUTPUT
+          echo "CHANGELOG_UPDATED=false" >> $GITHUB_OUTPUT
+          for file in ${{ steps.changed-files.outputs.all_changed_files }}; do
+            echo $file
+            if [[ $file == "src/"* || $file == "tests/"* || $file == "requirements-minimal.txt" || $file == "requirements-testing.txt" || $file == "setup.py" || $file == ".github/"* ]]
```

```suggestion
            if [[ $file == "src/"* || $file == "tests/"* || $file == ".github/"* ]]
```

Shortening; the pyproject approach really saves us a lot of space